pkg/local_object_storage/metabase/metadata.go

```diff
@@ -147,7 +158,89 @@ func deleteMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID) error {
 	for kIDAttr, _ := c.Seek(pref); bytes.HasPrefix(kIDAttr, pref); kIDAttr, _ = c.Next() {
 		sepInd := bytes.LastIndex(kIDAttr, objectcore.MetaAttributeDelimiter)
 		if sepInd < 0 {
-			return fmt.Errorf("invalid key with prefix 0x%X in meta bucket: missing delimiter", kIDAttr[0])
+			return 0, fmt.Errorf("invalid key with prefix 0x%X in meta bucket: missing delimiter", kIDAttr[0])
+		}
+		attrK := kIDAttr[1+oid.Size : sepInd]
+		attrV := kIDAttr[sepInd+len(objectcore.MetaAttributeDelimiter):]
+		switch kStr := string(attrK); kStr {
+		case object.FilterType:
+			var typ object.Type
+			if !isParent && typ.DecodeString(string(attrV)) {
+				err := delUniqueIndexes(tx, cnr, id, typ, isParent)
+				if err != nil {
+					return 0, err
+				}
+			}
+		case object.FilterOwnerID:
+			if len(attrV) != user.IDSize {
+				return 0, fmt.Errorf("invalid owner index value length %d", len(attrV))
+			}
+			owner := user.ID(attrV)
+			err = delFKBTIndexItem(tx, namedBucketItem{
+				name: ownerBucketName(cnr, bktKeyBuff),
+				key:  []byte(owner.EncodeToString()),
+				val:  id[:],
+			})
+			if err != nil {
+				return 0, err
+			}
+		case object.FilterPayloadChecksum:
+			err = delListIndexItem(tx, namedBucketItem{
+				name: payloadHashBucketName(cnr, bktKeyBuff),
+				key:  attrV,
+				val:  id[:],
+			})
+			if err != nil {
+				return 0, err
+			}
+		case object.FilterParentID:
+			if len(attrV) == oid.Size {
+				parent = oid.ID(attrV)
+			}
+			delUniqueIndexItem(tx, namedBucketItem{
+				name: parentBucketName(cnr, bktKeyBuff),
+				key:  attrV,
+			})
+			err = delListIndexItem(tx, namedBucketItem{
+				name: parentBucketName(cnr, bktKeyBuff),
+				key:  attrV,
+				val:  id[:],
+			})
+			if err != nil {
+				return 0, err
+			}
+		case object.FilterSplitID:
+			err = delListIndexItem(tx, namedBucketItem{
+				name: splitBucketName(cnr, bktKeyBuff),
+				key:  attrV,
+				val:  id[:],
+			})
+			if err != nil {
+				return 0, err
+			}
+		case object.FilterFirstSplitObject:
+			err = delListIndexItem(tx, namedBucketItem{
+				name: firstObjectIDBucketName(cnr, bktKeyBuff),
+				key:  attrV,
+				val:  id[:],
+			})
+			if err != nil {
+				return 0, err
+			}
+		case object.FilterPayloadSize:
+			size, _ = strconv.ParseUint(string(attrV), 10, 64)
+		default:
+			const systemFieldPrefix = "$Object:"
+			if !bytes.HasPrefix(attrK, []byte(systemFieldPrefix)) {
```

Comment:
else worth log, normally we dont expect sys attrs here

Reply:
not all cases are provided in this `switch`, there will always be some other fields. it is not an error check, it is just a not-trigger-when-not-needed check

Reply:
u mean not all possible old indexes are cased in the switch now? They all should be covered

Reply:
not all newly indexed info was indexed in the old indexes. i carefully looked at the code and deleted only things that were deleted before; no functional changes